test/unit/api/memory/aligned_ptr.cpp

```diff
@@ -170,3 +170,20 @@ TTS_CASE("aligned_ptr provides pointer-like interface")
     }
   }
 }
+
+TTS_CASE("previous aligned address")
+{
+  alignas(16) std::array<std::byte, 16> data;
```

Comment:
this array should be more than 16 in size and check for every 16 elements. This way the over aligned stack variable won't cause issues.